custom/text-chat/contexts/ChatProvider.js

```diff
@@ -6,61 +6,73 @@ import React, {
   useState,
 } from 'react';
 import { useCallState } from '@custom/shared/contexts/CallProvider';
+import { useSharedState } from '@custom/shared/hooks/useSharedState';
 import { nanoid } from 'nanoid';
 import PropTypes from 'prop-types';
 
 export const ChatContext = createContext();
 
 export const ChatProvider = ({ children }) => {
   const { callObject } = useCallState();
-  const [chatHistory, setChatHistory] = useState([]);
+  const { sharedState, setSharedState } = useSharedState({
+    initialValues: {
+      chatHistory: [],
+    },
+    broadcast: false,
+  });
   const [hasNewMessages, setHasNewMessages] = useState(false);
 
   const handleNewMessage = useCallback(
     (e) => {
+      if (e?.data?.message?.type) return;
       const participants = callObject.participants();
       const sender = participants[e.fromId].user_name
         ? participants[e.fromId].user_name
         : 'Guest';
 
-      setChatHistory((oldState) => [
-        ...oldState,
-        { sender, message: e.data.message, id: nanoid() },
-      ]);
+      setSharedState(values => ({
+        ...values,
+        chatHistory: [
+          ...values.chatHistory,
+          { sender, senderID: e.fromId, message: e.data.message, id: nanoid() },
+        ]
+      }));
 
       setHasNewMessages(true);
     },
-    [callObject]
+    [callObject, setSharedState]
   );
 
+
   const sendMessage = useCallback(
     (message) => {
-      if (!callObject) {
-        return false;
-      }
+      if (!callObject) return;
 
       console.log('💬 Sending app message');
 
       callObject.sendAppMessage({ message }, '*');
 
+      const participants = callObject.participants();
       // Get the sender (local participant) name
-      const sender = callObject.participants().local.user_name
-        ? callObject.participants().local.user_name
+      const sender = participants.local.user_name
+        ? participants.local.user_name
         : 'Guest';
-
-      // Update local chat history
-      return setChatHistory((oldState) => [
-        ...oldState,
-        { sender, message, id: nanoid(), isLocal: true },
-      ]);
+      const senderID = participants.local.user_id;
+
+      // Update shared state chat history
+      return setSharedState({
+        ...sharedState,
+        chatHistory: [
+          ...sharedState.chatHistory,
+          { sender, senderID, message, id: nanoid() }
+        ]
+      });
```

Comment:
We could also avoid having to rely on `sharedState` here. Your call ☺️

Reply:
yeah, updating it in a second :)

Reply:
Done, updated it.